### PR TITLE
Add support sparsity freeze epoch for magnitude sparsity.

### PR DESCRIPTION
**Pull request by asenina**

### CODE REVIEW

File: nncf/sparsity/layers.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ class BinaryMask(nn.Module):
     def __init__(self, size):
         super().__init__()
         self.register_buffer("_binary_mask", torch.ones(size))
+        self.sparsify = True
```
Comment (vshampor) [inline nncf/sparsity/layers.py:27]: Rename to `frozen` and update the logic accordingly

**Review by vshampor — CHANGES_REQUESTED** [review]

File: nncf/sparsity/layers.py (hunk 1 above)
Comment (asenina) [reply nncf/sparsity/layers.py:27]: @vshampor  Ok, but then class `RBSparsifyingWeight` will have two attributes `sparsify` and `frozen`. It looks redundant.

Comment (vshampor) [reply nncf/sparsity/layers.py:27]: Guess you'll have to rename the `RBSparsifytinWeight`'s `sparsify` attribute to `frozen` as well.

Comment (asenina) [reply nncf/sparsity/layers.py]: Done.

**Review by vshampor — APPROVED** [review]

**Pull request merged**